File: mtgSearch/src/main/java/com/dbottillo/mtgsearchfree/view/CardsView.java
package com.dbottillo.mtgsearchfree.view;

import com.dbottillo.mtgsearchfree.model.CardsBucket;
import com.dbottillo.mtgsearchfree.model.DeckBucket;
import com.dbottillo.mtgsearchfree.ui.BasicView;

public interface CardsView extends BasicView {

    void cardsLoaded(CardsBucket bucket);

    void deckLoaded(DeckBucket bucket);

    void favIdLoaded(int[] favourites);

    void cardTypePreferenceChanged(boolean grid);

}

File: mtgSearch/src/main/kotlin/com/dbottillo/mtgsearchfree/view/CardsView.kt
package com.dbottillo.mtgsearchfree.view

import com.dbottillo.mtgsearchfree.resources.CardsBucket
import com.dbottillo.mtgsearchfree.resources.MTGCard
import java.util.*

interface CardsView : BasicView {

    fun cardLoaded(bucket: CardsBucket)

    fun luckyCardsLoaded(cards: ArrayList<MTGCard>)

    fun favIdLoaded(favourites: IntArray)
}